libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,344 @@
+import {
+  InvalidContextError,
+  ParseError,
+  StandardResolutionReasons,
+  TargetingKeyMissingError,
+} from '@openfeature/core';
+import { isResolutionError, ResolutionError } from './model/resolution-error';
+import { BulkEvaluationStatus, EvaluateFlagsResponse } from './model/evaluate-flags-response';
+
+export class OfrepWebProvider implements Provider {
+  DEFAULT_POLL_INTERVAL = 30000;
+
+  metadata = {
+    name: OfrepWebProvider.name,
+  };
+  readonly runsOn = 'client';
+  events = new OpenFeatureEventEmitter();
+  hooks?: Hook[] | undefined;
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+  // _options is the options used to configure the provider.
+  private _options: OfrepWebProviderOptions;
+
+  private _ofrepAPI: OFREPApi;
+  private _etag: string | null;
+  private _pollingInterval: number;
+  private _retryPollingAfter: Date | undefined;
+  private _cache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError } = {};
+  private _context: EvaluationContext | undefined;
+  private _pollingIntervalId?: number;
+
+  constructor(options: OfrepWebProviderOptions, logger?: Logger) {
+    this._options = options;
+    this._logger = logger;
+    this._etag = null;
+    this._ofrepAPI = new OFREPApi(this._options.baseUrl);
+    this._pollingInterval = this._options.pollInterval ?? this.DEFAULT_POLL_INTERVAL;
+  }
+
+  resolveBooleanEvaluation(flagKey: string): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolean');
+  }
+  resolveStringEvaluation(flagKey: string): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'string');
+  }
+  resolveNumberEvaluation(flagKey: string): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'number');
+  }
+  resolveObjectEvaluation<T extends JsonValue>(flagKey: string): ResolutionDetails<T> {
+    return this.evaluate(flagKey, 'object');
+  }
+
+  /**
+   * onContextChange is called when the context changes, it will re-evaluate the flags with the new context
+   * and update the cache.
+   * @param oldContext - the old evaluation context
+   * @param newContext - the new evaluation context
+   */
+  async onContextChange(oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    try {
+      this._context = newContext;
+
+      const now = new Date();
+      if (this._retryPollingAfter !== undefined && this._retryPollingAfter > now) {
+        // we do nothing because we should not call the endpoint
+        return;
+      }
+
+      await this._evaluateFlags(newContext);
+    } catch (error) {
+      if (error instanceof OFREPApiTooManyRequestsError) {
+        this.events?.emit(ClientProviderEvents.Stale, { message: `${error.name}: ${error.message}` });
+        return;
+      }
+
+      if (
+        error instanceof OpenFeatureError ||
+        error instanceof OFREPApiFetchError ||
+        error instanceof OFREPApiUnauthorizedError ||
+        error instanceof OFREPForbiddenError
```

Comment:
I don't think this will be an issue here because we should be able to bundle everything in the provider when we publish. We shouldn't even need to publish ofrep core.